packages/xo-server/src/api/vm.mjs

```diff
@@ -1601,32 +1601,52 @@ createInterface.resolve = {
 
 // -------------------------------------------------------------------
 
-export async function attachPci({ vm, pciId }) {
-  await this.getXapiObject(vm).update_other_config('pci', pciId)
-}
+// https://docs.xcp-ng.org/compute/#5-put-this-pci-device-into-your-vm
+const getRawPciIds = pciIds => '0/'.concat(pciIds.join(',0/'))
```

Comment:
For me this is the inverse of raw.

IMHO, it would be easier to understand if there were two functions: `parsePciIds` and `formatPciIds`.

Reply:
Also, maybe you could parse it directly in `xapi-objects-to-xo` to remove the logic from `xo-web`.